src/Features/Onboarding.php

```diff
@@ -861,15 +854,23 @@ public static function reset_task_list() {
 			return;
 		}
 
-		$task_list_hidden = 1 === absint( $_GET['reset_task_list'] ) ? 'no' : 'yes'; // phpcs:ignore CSRF ok.
-		update_option( 'woocommerce_task_list_hidden', $task_list_hidden );
+		$task_list = TaskLists::get_list( 'setup' );
+
+		if ( ! $task_list ) {
+			return;
+		}
+		$show   = 1 === absint( $_GET['reset_task_list'] );
+		$update = $show ? $task_list->show() : $task_list->hide(); // phpcs:ignore CSRF ok.
+
+		if ( $update ) {
+			wc_admin_record_tracks_event(
+				'tasklist_toggled',
```

Comment:
It would be nice to be consistent with whether we treat "tasklist" as a single word, or as two words ("task list"). We have variable names and tracks event names that do it differently. This may make it more likely to introduce a typo in the future that would cause a defect.

Reply:
It would be nice.  I think two words is what we've settled on, but unfortunately this is a pre-existing tracks event so we should not change the name here.

Reply:
True.